code/modules/lighting/lighting_system.dm

```diff
@@ -58,6 +58,8 @@
 		remove_effect()
 		owner.light = null
 		owner = null
+	if(changed)
+		SSlighting.changed_lights -= src
```

Comment:
needs an error message. removing it from the list is obviously good but it would be nice to KNOW if there's a null in the list.